src/test/java/com/microsoft/sqlserver/jdbc/bulkCopy/BulkCopySendTemporalDataTypesAsStringTest.java

```diff
@@ -0,0 +1,225 @@
+ */
+package com.microsoft.sqlserver.jdbc.bulkCopy;
+
+import static org.junit.Assert.fail;
+
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.InputStreamReader;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Arrays;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+import com.microsoft.sqlserver.jdbc.ComparisonUtil;
+import com.microsoft.sqlserver.jdbc.RandomUtil;
+import com.microsoft.sqlserver.jdbc.SQLServerBulkCSVFileRecord;
+import com.microsoft.sqlserver.jdbc.SQLServerBulkCopy;
+import com.microsoft.sqlserver.jdbc.SQLServerDataSource;
+import com.microsoft.sqlserver.jdbc.TestUtils;
+import com.microsoft.sqlserver.testframework.AbstractSQLGenerator;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+import com.microsoft.sqlserver.testframework.PrepUtil;
+
+
+/**
+ * Test connection property sendTemporalDataTypesAsStringForBulkCopy
+ * This connection string, when set to FALSE, will send DATE, DATETIME, DATIMETIME2 DATETIMEOFFSET, SMALLDATETIME, and
+ * TIME
+ * datatypes as their respective types instead of sending them as String.
+ * Additionally, even without setting this connection string to FALSE, MONEY and SMALLMONEY datatypes will be
+ * sent as MONEY / SMALLMONEY datatypes instead of DECIMAL after these changes.
+ * 
+ * Note that with this connection property set to FALSE, the driver will only accept the default string literal format
+ * of each temporal datatype, for example:
+ * 
+ * DATE: YYYY-MM-DD
+ * DATETIME: YYYY-MM-DD hh:mm:ss[.nnn]
+ * DATETIME2: YYYY-MM-DD hh:mm:ss[.nnnnnnn]
+ * DATETIMEOFFSET: YYYY-MM-DD hh:mm:ss[.nnnnnnn] [{+|-}hh:mm]
+ * SMALLDATETIME:YYYY-MM-DD hh:mm:ss
+ * TIME: hh:mm:ss[.nnnnnnn]
+ * 
+ */
+@RunWith(JUnitPlatform.class)
+public class BulkCopySendTemporalDataTypesAsStringTest extends AbstractTest {
+    static String inputFile = "BulkCopyCSVSendTemporalDataTypesAsStringForBulkCopy.csv";
+    static String encoding = "UTF-8";
+    static String delimiter = ",";
+
+    static String destTableName = AbstractSQLGenerator
+            .escapeIdentifier(RandomUtil.getIdentifier("sendTemporalDataTypesAsStringForBulkCopyDestTable"));
+    static String destTableName2 = AbstractSQLGenerator
+            .escapeIdentifier(RandomUtil.getIdentifier("sendTemporalDataTypesAsStringForBulkCopyDestTable2"));
+    static String filePath = null;
+
+    /**
+     * Test basic case with sendTemporalDataTypesAsStringForBulkCopy connection property.
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testSendTemporalDataTypesAsStringForBulkCopy() throws SQLException {
+        beforeEachSetup();

```

Comment:
why not just use @BeforeEach?

Reply:
Sometimes beforeEach didn't work for me properly, so I just call beforeEachSetup. There shouldn't be any functional differences.